testing.py

```diff
@@ -968,6 +976,161 @@ def sample_cover_representation(self, name):
             media_type="image/png", content=open(sample_cover_path).read())[0]
 
 
+class SearchClientForTesting(ExternalSearchIndex):
+    """When creating an index, limit it to a single shard and disable
+    replicas.
+
+    This makes search results more predictable.
+    """
+
+    def setup_index(self, new_index=None):
+        return super(SearchClientForTesting, self).setup_index(
+            new_index, number_of_shards=1, number_of_replicas=0
+        )
+
+
+class ExternalSearchTest(DatabaseTest):
+    """
+    These tests require elasticsearch to be running locally. If it's not, or there's
+    an error creating the index, the tests will pass without doing anything.
+
+    Tests for elasticsearch are useful for ensuring that we haven't accidentally broken
+    a type of search by changing analyzers or queries, but search needs to be tested manually
+    to ensure that it works well overall, with a realistic index.
+    """
+
+    def setup(self):
+        super(ExternalSearchTest, self).setup(mock_search=False)
+
+        self.integration = self._external_integration(
+            ExternalIntegration.ELASTICSEARCH,
+            goal=ExternalIntegration.SEARCH_GOAL,
+            url=u'http://localhost:9200',
+            settings={
+                ExternalSearchIndex.WORKS_INDEX_PREFIX_KEY : u'test_index',
+                ExternalSearchIndex.TEST_SEARCH_TERM_KEY : u'test_search_term',
+            }
+        )
+
+        try:
+            self.search = SearchClientForTesting(self._db)
+        except Exception as e:
+            self.search = None
+            logging.error(
+                "Unable to set up elasticsearch index, search tests will be skipped.",
+                exc_info=e
+            )
+
+    def teardown(self):
+        if self.search:
+            if self.search.works_index:
+                self.search.indices.delete(self.search.works_index, ignore=[404])
+            self.search.indices.delete('the_other_index', ignore=[404])
+            self.search.indices.delete('test_index-v100', ignore=[404])
+            self.search.indices.delete('test_index-v9999', ignore=[404])
```

Comment:
I'm not sure which test creates these but it might make sense to remove them there instead of the parent class?

Reply:
It looks like -v100 wasn't being used anymore. I added a helper method that creates the index and registers it so it can be deleted during teardown.